test/functional/lua/diagnostic_spec.lua

```diff
@@ -1430,6 +1430,94 @@ describe('vim.diagnostic', function()
         eq(true, exec_lua('return _G.jumped'))
       end)
     end)
+
+    it('finds next diagnostic at a logical location after text changes before', function()
```

Comment:
is it possible for these tests to re-use a shared function, to make them less verbose?

indirection can make things harder to read, but if these tests are mostly doing the same kind of thing, then using a shared function can be much more readable because it makes it obvious what is being tested in each case.

Reply:
reused it via `describe` + `before_each`